pkg/openstack/loadbalancer.go

```diff
@@ -973,20 +1168,56 @@ func (lbaas *LbaasV2) getServiceAddress(clusterName string, service *corev1.Serv
 				PortID:            portID,
 				Description:       fmt.Sprintf("Floating IP for Kubernetes external service %s from cluster %s", serviceName, clusterName),
 			}
-			if svcConf.lbPublicSubnetID != "" {
-				floatIPOpts.SubnetID = svcConf.lbPublicSubnetID
-			}
-			if loadBalancerIP != "" {
-				floatIPOpts.FloatingIP = loadBalancerIP
-			}
 
-			klog.V(4).Infof("Creating floating ip with opts %+v", floatIPOpts)
+			if loadBalancerIP == "" && svcConf.lbPublicSubnetSpec.MatcherConfigured() {
```

Comment:
The condition `loadBalancerIP == ""` is always true here, please remove.

Reply:
I'm not sure about the control flow outside the function, but here it might be possible for `loadbalancerIP` to be empty. This is because the earlier `if` block `if floatIP == nil && loadBalancerIP != "" {` might be left without a return. Therefore the following `if` block `if floatIP == nil {` might be entered with a non-empty `loadbalancerIP`.